synthea/qpp_qicore.yaml

```diff
@@ -0,0 +1,156 @@
+            location: ServiceRequest.authoredOn
+            value: $getField([Procedure.performed]) #datetime
+          - location: ServiceRequest.code
+            value: $getField([Procedure.code])
+        writeback:
+          - location: Procedure.basedOn.reference
+            value: $setRef([ServiceRequest])
+      - resourceType: DeviceRequest
+        based_on:
+          resource: Device
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-devicerequest
+        fields:
+          - location: DeviceRequest.code.reference
+            value: $findRef([Device])
+          - location: DeviceRequest.authoredOn
+            value: $getField([Device.manufactureDate]) # manufacture time is set 3 weeks before device model's start, so this is close enough
+          - location: DeviceRequest.status
+            value: completed
+          - location: DeviceRequest.intent
+            value: order
+      # Creating one AdverseEvent per patient
+      - resourceType: AdverseEvent
+        based_on:
+          resource: Patient
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-adverseevent
+        fields:
+          - location: AdverseEvent.event.coding
+            value: $randomCode([http://hl7.org/fhir/ValueSet/adverse-event-type]) # may change in future to use a smaller subset of the ValueSet
+      # Creating one Communication per patient
+      - resourceType: Communication
+        based_on:
+          resource: Patient
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-communication
+        fields:
+          - location: Communication.status
+            value: $randomCode([http://hl7.org/fhir/ValueSet/event-status,code])
+      # Creating one MedicationAdministration for each MedicationRequest
+      - resourceType: MedicationAdministration
+        based_on:
+          resource: MedicationRequest
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-medicationadministration
+        fields:
+          - location: MedicationAdministration.status
+            value: completed
+          - location: MedicationAdministration.medication.coding
+            value: $randomCode([http://hl7.org/fhir/ValueSet/medication-codes])
+          - location: MedicationAdministration.effective
+            value: $getField([MedicationRequest.dispenseRequest.validityPeriod])
+      # Creating one MedicationDispense for each MedicationRequest resource
+      - resourceType: MedicationDispense
+        based_on:
+          resource: MedicationRequest
+          profiles:
+            - http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-medicationdispense
+        fields:
+          - location: MedicationDispense.status
+            value: $randomCode([http://hl7.org/fhir/ValueSet/medicationdispense-status,code])
```

Comment:
Just a note for the future work if selecting from a list of codes becomes an available feature... because the MedicationRequest will only be `active` or `stopped`, we could potentially map (using the new `if` feature) `active` to the positive status values for dispense (preparation, in-progress, on-hold, completed) and map `stopped` to the negative values (cancelled, entered-in-error, stopped, declined).